### PR TITLE
unigine-superposition: init at 1.1

### PR DESCRIPTION
**Pull request by barinov274**
I want to add The Unigine Superposition GPU benchmarking tool. People will be able to install this benchmarking tool on nixos and benchmark their GPU.

- Built on platform(s)
  - [x] x86_64-linux

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13971,4 +13971,10 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  BarinovMaxim = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13974]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  libX11,
+  libICE,
+  libXrender,
+  autoPatchelfHook,
+  makeWrapper,
+  mkDerivation,
+  xkeyboard_config,
+  fetchurl,
+  buildFHSUserEnv,
+  openal
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:26]:
```suggestion
{ lib
, glib
, stdenv
, dbus
, freetype
, fontconfig
, zlib
, qtquickcontrols2
, libXinerama
, libxcb
, libSM
, libXi
, libglvnd
, libXext
, libXrandr
, mailspring
, libX11
, libICE
, libXrender
, autoPatchelfHook
, makeWrapper
, mkDerivation
, xkeyboard_config
, fetchurl
, buildFHSUserEnv
, openal
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    echo "Icon=Superposition" >> $DESKTOP
+    echo "Terminal=false" >> $DESKTOP
+    echo "Categories=Launcher;Benchmark;" >> $DESKTOP
+
+    # install Superposition.desktop and icon
+    cp $INSTALL_DIR/Superposition.png $out/share/icons/
+    for RES in 16 24 32 48 64 128 256
+    do
+        mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
+        cp $INSTALL_DIR/icons/superposition_icon_$RES.png $out/share/icons/hicolor/"$RES"x"$RES"/apps/Superposition.png
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:151]:
```suggestion
    for RES in 16 24 32 48 64 128 256; do
      mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
      cp $INSTALL_DIR/icons/superposition_icon_$RES.png $out/share/icons/hicolor/"$RES"x"$RES"/apps/Superposition.png
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    # create .desktop file
+    echo "[Desktop Entry]" > $DESKTOP
+    echo "Type=Application" >> $DESKTOP
+    echo "Exec=$EXEC" >> $DESKTOP
+    echo "Path=$INSTALL_DIR/bin" >> $DESKTOP
+    echo "Name=Superposition Benchmark" >> $DESKTOP
+    echo "GenericName=A GPU Stress test tool from the UNIGINE" >> $DESKTOP
+    echo "Icon=Superposition" >> $DESKTOP
+    echo "Terminal=false" >> $DESKTOP
+    echo "Categories=Launcher;Benchmark;" >> $DESKTOP
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:144]: Please use createDesktopItem.

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+   extraInstallCommands = ''
+    VERSION=$version
+    INSTALL_DIR=${superposition}/lib/unigine/superposition/
+    EXEC=$out/bin/Superposition
+    DESKTOP=$out/share/applications/Superposition.desktop
+
+    # create directories
+    mkdir -p $out/share/icons/hicolor
+    mkdir -p $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:133]:
```suggestion
    mkdir -p $out/share/applications $out/share/icons/hicolor
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+     libXrender
+     openal
+   ];
+   runScript = "superposition";
+
+   extraInstallCommands = ''
+    VERSION=$version
+    INSTALL_DIR=${superposition}/lib/unigine/superposition/
+    EXEC=$out/bin/Superposition
+    DESKTOP=$out/share/applications/Superposition.desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:129]:
```suggestion
```
Just inline this.

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  '';
+};
+
+in
+
+ /*
+  * We can patch the "/bin/superposition", but "/bin/launcher" checks it for changes.
+  * For that we need use a buildFHSUserEnv.
+  * FHS simulates a classic linux shell
+  */
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:96]:
```suggestion
  // We can patch the "/bin/superposition", but "/bin/launcher" checks it for changes.
  // For that we need use a buildFHSUserEnv.
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+    echo "exec $out/lib/unigine/superposition/Superposition" >> $out/bin/superposition
+    chmod +x $out/bin/superposition
+
+    wrapProgram $out/lib/unigine/superposition/Superposition --set QT_XKB_CONFIG_ROOT ${xkeyboard_config} --run "cd $out/lib/unigine/superposition/"
+  '';
+
+    dontUnpack = true;
+    dontConfigure = true;
+    dontBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:82]: Do these break something? If not please enable them again.

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    dontWrapQtApps=true;
+
+  postPatchMkspecs = ''
+    cp -f $name/bin/superposition $out/lib/unigine/superposition/bin/superposition #cp unpatched "superposition"
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:86]:
```suggestion
    cp -f $name/bin/superposition $out/lib/unigine/superposition/bin/superposition
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+    mkdir -p $out/lib/unigine/superposition/
+    mkdir $out/bin
+
+    cp -r $name/* $out/lib/unigine/superposition/
+
+    echo "exec $out/lib/unigine/superposition/Superposition" >> $out/bin/superposition
+    chmod +x $out/bin/superposition
+
+    wrapProgram $out/lib/unigine/superposition/Superposition --set QT_XKB_CONFIG_ROOT ${xkeyboard_config} --run "cd $out/lib/unigine/superposition/"
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:77]:
```suggestion
    wrapProgram $out/lib/unigine/superposition/Superposition \
      --set QT_XKB_CONFIG_ROOT ${xkeyboard_config} \
      --chdir $out/lib/unigine/superposition/
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    libX11
+    libICE
+    libXrender
+  ];
+
+  installPhase = ''
+    sh $src --target $name --noexec
+
+    mkdir -p $out/lib/unigine/superposition/
+    mkdir $out/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:70]:
```suggestion
    mkdir -p $out/bin $out/lib/unigine/superposition/
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+    libXext
+    libXrandr
+    mailspring
+    libX11
+    libICE
+    libXrender
+  ];
+
+  installPhase = ''
+    sh $src --target $name --noexec
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:67]:
```suggestion
    bash $src --target $name --noexec
```

**Review by SuperSandro2000 — COMMENTED** [review]
Also please format with nixpkgs-fmt.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (barinov274) [reply maintainers/maintainer-list.nix:13974]: It's already in separated commit

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 7 above)
Comment (barinov274) [reply pkgs/applications/graphics/unigine-superposition/default.nix:96]: error: syntax error, unexpected UPDATE, at /home/max/Документы/nixpkgs/pkgs/applications/graphics/unigine-superposition/default.nix:104:1

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 10 above)
Comment (barinov274) [reply pkgs/applications/graphics/unigine-superposition/default.nix:77]: Builder called die: makeWrapper doesn't understand the arg --chdir

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 4 above)
Comment (barinov274) [reply pkgs/applications/graphics/unigine-superposition/default.nix]: maybe makeDesktopItem?

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:13974]: but not sorted alphabetically.

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/unigine-superposition/default.nix]: Both actually.

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 7 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/unigine-superposition/default.nix]: `#` is for comments in nix.

File: pkgs/applications/graphics/unigine-superposition/default.nix (hunk 10 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/unigine-superposition/default.nix]: Than that is still in staging.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (barinov274) [reply maintainers/maintainer-list.nix]: It isn't important now. Because it's merged https://github.com/NixOS/nixpkgs/pull/164121

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  extraInstallCommands = ''
+    # create directories
+    mkdir -p $out/share/icons/hicolor $out/share/applications
+    # create .desktop file
+    ln -s ${desktopItem}/share/applications/* $out/share/applications
+    # install Superposition.desktop and icon
+    cp ${superposition}/lib/unigine/superposition/Superposition.png $out/share/icons/
+    for RES in 16 24 32 48 64 128 256; do
+        mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
+        cp ${superposition}/lib/unigine/superposition/icons/superposition_icon_$RES.png $out/share/icons/hicolor/"$RES"x"$RES"/apps/Superposition.png
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:135]:
```suggestion
      mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
      cp ${superposition}/lib/unigine/superposition/icons/superposition_icon_$RES.png $out/share/icons/hicolor/"$RES"x"$RES"/apps/Superposition.png
```

File: pkgs/applications/graphics/unigine-superposition/default.nix [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+
+    installPhase = ''
+      bash $src --target $name --noexec
+      mkdir -p $out/bin $out/lib/unigine/superposition/
+      cp -r $name/* $out/lib/unigine/superposition/
+      echo "exec $out/lib/unigine/superposition/Superposition" >> $out/bin/superposition
+      chmod +x $out/bin/superposition
+       wrapProgram $out/lib/unigine/superposition/Superposition \
+      --set QT_XKB_CONFIG_ROOT ${xkeyboard_config} \
+      --run "cd $out/lib/unigine/superposition/"
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-superposition/default.nix:75]:
```suggestion
        --set QT_XKB_CONFIG_ROOT ${xkeyboard_config} \
        --run "cd $out/lib/unigine/superposition/"
```
